doc/source/api.rst

```diff
@@ -1,17 +1,111 @@
 Python & R API
 ==============
 
-The application programming interface (API) for |MESSAGEix| model developers
-is implemented in Python. Support for R usage of the core classes is provided
-through the `reticulate`_ package. For instance::
+The application programming interface (API) for |MESSAGEix| model developers is implemented in Python:
+
+.. contents::
+   :local:
+
+Support for R usage of the core classes is provided through the `reticulate`_ package. For instance::
 
     > library(reticulate)
-    > message <- library("message_ix")
-    > s <- messageix$Scenario(…)
+    > ixmp <- import('ixmp')
+    > message_ix <- import('message_ix')
+    > mp <- ixmp$Platform(...)
+    > scen <- message_ix$Scenario(mp, ...)
 
 .. _`reticulate`: https://rstudio.github.io/reticulate/
 
-.. toctree::
 
-   api/ixmp.rst
-   api/message_ix.rst
+``ixmp`` package
+----------------
+
+:mod:`ixmp` provides three classes. These are fully described by the :doc:`ixmp documentation <ixmp:index>`, which is cross-linked from many places in the |MESSAGEix| documentation.
+
+.. autosummary::
+
+   ~ixmp.Platform
+   ~ixmp.TimeSeries
+   ~ixmp.Scenario
+
+:mod:`ixmp` also provides some utility classes and methods:
+
+.. autosummary::
+
+   ixmp.config.Config
+   ixmp.model.MODELS
+   ixmp.model.get_model
+   ixmp.testing.make_dantzig
+
+
+.. currentmodule:: message_ix
+
+``message_ix`` package
+----------------------
+
+|MESSAGEix| models are created using the :py:class:`message_ix.Scenario` class. Several utility methods are also provided in the module :py:mod:`message_ix.utils`.
+
+.. autoclass:: message_ix.Scenario
+   :members:
+   :show-inheritance:
+   :inherited-members:
+
+   This class extends :class:`ixmp.Scenario` and :class:`ixmp.TimeSeries` and
+   inherits all their methods. Documentation of these inherited methods is
+   included here for convenience. :class:`message_ix.Scenario` defines
+   additional methods specific to |MESSAGEix|:
+
+   .. autosummary::
+
+      add_cat
+      add_horizon
+      add_spatial_sets
+      cat
+      cat_list
+      firstmodelyear
+      read_excel
+      rename
+      to_excel
+      vintage_and_active_years
+      years_active
+
+
+
+Model classes
+-------------
+
+.. currentmodule:: message_ix.models
+
+.. automodule:: message_ix.models
+   :exclude-members: MESSAGE, MESSAGE_MACRO
+
+.. autodata:: DEFAULT_CPLEX_OPTIONS
+
+   These configure the GAMS CPLEX solver (or another solver, if selected); see `the solver documentation <https://www.gams.com/latest/docs/S_CPLEX.html>`_ for possible values.
+
+.. autoclass:: MESSAGE
+   :members:
+   :show-inheritance:
+
+   The MESSAGE Python class encapsulates the GAMS code for the core MESSAGE mathematical formulation.
+   The *model_options* arguments are received from :meth:`.Scenario.solve`, and—except for *solve_options*—are passed on to the parent class :class:`~ixmp.model.gams.GAMSModel`; see there for a full list of options.
+
+.. autoclass:: MESSAGE_MACRO
```

Comment:
This looks to be broken on RTD? https://message-ix-khaeru.readthedocs.io/en/feature-ixmp-backend/api.html#message_ix.models.MESSAGE.defaults

Reply:
Good eye. The code is working as intended, but that's indeed not what we want the user to be seeing in the documentation. I'll try for a fix.